files/default/audit_report.rb

```diff
@@ -94,12 +94,26 @@ def call(reporter)
         runner.run
       end
 
+      # extracts relevant node data
+      def gather_nodeinfo
+        n = run_context.node
+        {
+          node: n.name,
+          os: {
+            # arch: n['arch'],
+            release: n['platform_version'],
+            family: n['platform'],
+          },
+          environment: n.environment,
+        }
+      end
+
       def send_report(reporter, server, user)
         Chef::Log.info "Reporting to #{reporter}"
 
         # TODO: harmonize reporter interface
         if reporter == 'chef-visibility'
-          Collector::ChefVisibility.new(entity_uuid, run_id, run_context.node.name).send_report
+          Collector::ChefVisibility.new(entity_uuid, run_id, gather_nodeinfo).send_report
 
```

Comment:
i think only node name was being sent to viz, which is why the tests are complaining